conductor-client/src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -143,60 +142,67 @@ class ExternalDatabaseManagementService(
-            dbCredentialService.getDbUsername(it.key)
+            es.flags.remove(EntitySetFlag.TRANSPORTED)
+            entitySets.set(entitySetId, es)
+        } finally {
+            entitySets.unlock(entitySetId)
         }
+    }
 
-        val entityTypeId = entitySets.submitToKey(
+    fun transportEntitySet(organizationId: UUID, entitySetId: UUID) {
+        val ptIds = entitySets.submitToKey(
                 entitySetId,
                 GetEntityTypeFromEntitySetEntryProcessor()
-        )
-
-        val accessCheckCompletion = entityTypeId.thenCompose { etid ->
-            entityTypes.getAsync(etid!!)
-        }.thenApplyAsync { entityType ->
-            entityType.properties.mapTo(mutableSetOf()) { ptid ->
-                AccessCheck(AclKey(entitySetId, ptid), EnumSet.of(Permission.READ))
-            }
-        }
-
-        val transporterColumnsCompletion = entityTypeId.thenCompose { etid ->
+        ).thenCompose { etid ->
             requireNotNull(etid) {
                 "Entity set $entitySetId has no entity type"
             }
             transporterState.submitToKey(etid, GetPropertyTypesFromTransporterColumnSetEntryProcessor())
-        }.thenCompose { transporterPtIds ->
+        }
+
+        val ptIdsToFqns = ptIds.thenCompose { transporterPtIds ->
             propertyTypes.submitToKeys(transporterPtIds, GetFqnFromPropertyTypeEntryProcessor())
+        }.thenApply { ptIdToFqn ->
+            ptIdToFqn.mapTo( mutableSetOf() ) { PropertyTypeIdFqn(it.key, it.value) }
         }
 
-        val userToPermissionsCompletion = accessCheckCompletion.thenApply { accessChecks ->
-            userToPrincipals.mapValues { (_, principals) ->
-                authorizationManager.accessChecksForPrincipals(accessChecks, principals).filter {
-                    it.permissions[Permission.READ]!!
-                }.map {
-                    it.aclKey[1]
-                }.toList()
+        val tableCols = ptIds.thenApply { transporterPtIds ->
+            transporterPtIds.associate {
+                AclKey(entitySetId, it) to TableColumn(organizationId, entitySetId, it)
             }
         }
 
-        userToPermissionsCompletion.thenCombine(transporterColumnsCompletion) { userToPtCols, transporterColumns ->
-            val userToEntitySetColumnNames = userToPtCols.mapValues { (_, columns) ->
-                columns.map {
-                    transporterColumns.get(it).toString()
-                }
+        val acls = permissions.entrySet(Predicates.and(
+                Predicates.equal<AceKey, AceValue>(PermissionMapstore.ROOT_OBJECT_INDEX, entitySetId),
+                Predicates.equal<AceKey, AceValue>(PermissionMapstore.SECURABLE_OBJECT_TYPE_INDEX, SecurableObjectType.PropertyTypeInEntitySet))
+        ).map { (key, valu) ->
+            Acl( key.aclKey, listOf( Ace(key.principal, valu as Set<Permission>, Optional.of(valu.expirationDate))) )
```

Comment:
might be ideal to group by aclKey so as to not have singleton ace lists, although idk if that's being done anyway down the line